source/NPC.cpp

```diff
@@ -241,6 +273,42 @@ void NPC::Save(DataWriter &out) const
 
 
 
+// Update to spawn conditions for if this NPC can be placed.
+void NPC::CanSpawn(const PlayerInfo &player)
+{
+	// Check the toSpawn conditions each time this function is called until they pass.
+	if(!passedSpawnConditions)
+		passedSpawnConditions = toSpawn.Test(player.Conditions());
```

Comment:
> Do empty conditions always test-evaluate to true? If not, then you would want to add toSpawn.IsEmpty() as a reason to set the variable to true.

~~Grumble grumble I had that and you made a comment about it.~~ Yes. Empty conditions are always true. In Mission::CanOffer, toOffer is always tested, even if empty, whereas toFail must not be empty (as otherwise an empty toFail would evaluate to true). Same with how I'm always testing toSpawn but only testing toDespawn if it's not empty.

https://github.com/endless-sky/endless-sky/blob/3d4a0b233dbd6a73b46285be64bd3052be49efed/source/Mission.cpp#L576-L580

I also skimmed through ConditionSet::Test, which seemed to back this up.

Reply:
Okay, then this code here is fine.